backend/src/db/providers/clientCourseProvider.ts

```diff
@@ -168,7 +178,7 @@ const arrangeAssessment = async (courseId: string) => {
   }
 };
 
-const getClientCoursesByCourseId = async (courseId: string) => {
+const getClientCoursesByCourseId = async (courseId: string): Promise<IClientCourse | null> => {
```

Comment:
I don't really like `| null` here. Do we have an opportunity to avoid it?

Reply:
from it depends other parts